src/WebJobs.Script/Diagnostics/FunctionTraceWriterFactory.cs

```diff
@@ -19,20 +20,21 @@ public FunctionTraceWriterFactory(ScriptHostConfiguration scriptHostConfig)
             _scriptHostConfig = scriptHostConfig;
         }
 
-        public TraceWriter Create(string functionName)
+        public TraceWriter Create(string functionName, ScriptType? scriptType = null)
```

Comment:
Rather than pass in ScriptType here, recommend just having a nullable logDirectory parameter that defaults to Function when not specified.